test/extensions/bootstrap/reverse_tunnel/downstream_socket_interface/reverse_connection_io_handle_test.cc

```diff
@@ -2882,6 +2900,164 @@ TEST_F(ReverseConnectionIOHandleTest, AcceptMethodSocketAndFdFailures) {
   }
 }
 
+// Tests the case where dynamic_cast succeeds and SSL_set_quiet_shutdown is called.
+TEST_F(ReverseConnectionIOHandleTest, OnConnectionDoneTlsConnectionQuietShutdownSuccess) {
+  setupThreadLocalSlot();
+
+  auto config = createDefaultTestConfig();
+  io_handle_ = createTestIOHandle(config);
+  EXPECT_NE(io_handle_, nullptr);
+
+  createTriggerPipe();
+  EXPECT_TRUE(isTriggerPipeReady());
+
+  // Set up mock thread local cluster.
+  auto mock_thread_local_cluster = std::make_shared<NiceMock<Upstream::MockThreadLocalCluster>>();
+  EXPECT_CALL(cluster_manager_, getThreadLocalCluster("test-cluster"))
+      .WillRepeatedly(Return(mock_thread_local_cluster.get()));
+
+  auto mock_priority_set = std::make_shared<NiceMock<Upstream::MockPrioritySet>>();
+  EXPECT_CALL(*mock_thread_local_cluster, prioritySet())
+      .WillRepeatedly(ReturnRef(*mock_priority_set));
+
+  auto host_map = std::make_shared<Upstream::HostMap>();
+  auto mock_host = createMockHost("192.168.1.1");
+  (*host_map)["192.168.1.1"] = std::const_pointer_cast<Upstream::Host>(mock_host);
+  EXPECT_CALL(*mock_priority_set, crossPriorityHostMap()).WillRepeatedly(Return(host_map));
+
+  addHostConnectionInfo("192.168.1.1", "test-cluster", 1);
+
+  // Create a mock connection with TLS.
+  auto mock_connection = setupMockConnection();
+
+  // Create a mock SSL object to verify SSL_set_quiet_shutdown is called.
+  SSL_CTX* ctx = SSL_CTX_new(TLS_method());
```

Comment:
nit: this can be `bssl::UniquePtr<SSL_CTX> ctx(SSL_CTX_new());`, so don't need to free later.